perfkitbenchmarker/providers/gcp/gcp_spanner.py

```diff
@@ -0,0 +1,128 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module containing class for GCP's spanner instances.
+
+Instances can be created and deleted.
+"""
+
+import json
+import logging
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import resource
+from perfkitbenchmarker.providers.gcp import util
+
+FLAGS = flags.FLAGS
+
+
+class GcpSpannerInstance(resource.BaseResource):
+  """Object representing a GCP Spanner Instance.
+
+  Attributes:
+    name: Instance name.
+    desp: Description of the instance.
+    nodes: Number of nodes in the instance.
+    config: config the instance.
+  """
+
+  def __init__(self, name, desp, nodes, config, database, ddl):
+    super(GcpSpannerInstance, self).__init__()
+    self._nodes = nodes
+    self._name = name
+    self._desp = desp
+    self._config = config
+    self._database = database
+    self._ddl = ddl
+
+    # Cloud Spanner does not explicitly set the following common flags.
+    self.project = util.GetDefaultProject()
+    self.zone = None
+
+  def _Create(self):
+    """Creates the instance, the database, and update the schema."""
+    cmd = util.GcloudCommand(self, 'spanner', 'instances', 'create', self._name)
+    cmd.flags['description'] = self._desp
+    cmd.flags['nodes'] = self._nodes
+    cmd.flags['config'] = self._config
+    _, _, retcode = cmd.Issue()
+    if retcode != 0:
+      logging.error('Create GCP Spanner instance failed.')
+      return
+
+    cmd = util.GcloudCommand(self, 'spanner', 'databases', 'create',
+                             self._database)
+    cmd.flags['instance'] = self._name
+    _, _, retcode = cmd.Issue()
+    if retcode != 0:
+      logging.error('Create GCP Spanner database failed.')
+      return
+
+    cmd = util.GcloudCommand(self, 'spanner', 'databases', 'ddl', 'update',
+                             self._database)
+    cmd.flags['instance'] = self._name
+    cmd.flags['ddl'] = self._ddl
+    _, _, retcode = cmd.Issue()
+    if retcode != 0:
+      logging.error('Update GCP Spanner database schema failed.')
+    else:
+      logging.info('Created GCP Spanner instance and database.')
+
+  def _Delete(self):
+    """Deletes the instance."""
+    cmd = util.GcloudCommand(self, 'spanner', 'instances', 'delete',
+                             self._name)
+    _, _, retcode = cmd.Issue()
+    if retcode != 0:
+      logging.error('Delete GCP Spanner instance failed.')
+    else:
+      logging.info('Deleted GCP Spanner instance.')
+
+  def _Exists(self):
+    """Returns true if the instance and the database exists."""
+    cmd = util.GcloudCommand(self, 'spanner', 'instances', 'list')
```

Comment:
$ gcloud spanner instances describe [MY_INSTANCE_ID]

Is there a reason for not using the specific api for the instance ?